olp-cpp-sdk-core/tests/cache/DefaultCacheImplTest.cpp

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include <gtest/gtest.h>
+
+#include <cache/DefaultCacheImpl.h>
+#include <olp/core/utils/Dir.h>
+
+using namespace olp::cache;
+
+class DefaultCacheImplHelper : public DefaultCacheImpl {
+ public:
+  DefaultCacheImplHelper(const CacheSettings& settings)
+      : DefaultCacheImpl(settings){};
+
+  bool HasLruCache() const { return GetMutableCacheLru().operator bool(); }
```

Comment:
This is very strange.
```suggestion
  bool HasLruCache() const { return GetMutableCacheLru() != nullptr; }
```
is probably what you want to do.